tests/Controller/ExportControllerTest.php

```diff
@@ -228,4 +214,189 @@ public function testCorrectTranslatedProductIsExported(): void
         $this->assertSame(1, (int)$parsedResponse->items->attributes()->count);
         $this->assertSame('FINDOLOGIC Product DE', $parsedResponse->items->item->names->name->__toString());
     }
+
+    /**
+     * Test ensures that the Shopware Router generates the product URLs based on the current language.
+     * This special case applies if a Sales Channel has multiple languages and the export is called from
+     * within a different Sales Channel. E.g.
+     *
+     * Sales Channel "Germany" can be accessed with URL https://some-shop.de, therefore the configured export
+     * URL is set to it. Now there is a second domain for "United Kingdom", which can be accessed with the
+     * URL https://some-shop.co.uk. This test ensures that the products will use the UK domain, when calling
+     * the URL of the German Sales Channel with the shopkey of the United Kingdom Sales Channel:
+     * https://some-shop.de/findologic?shopkey=from-united-kingdom => https://shome-shop.co.uk/detail/...
+     *
+     * Runs in a separate process to not interfere with other tests.
+     * @runInSeparateProcess
+     */
+    public function testProductsWithoutSeoUrlsWillExportTheUrlBasedOnTheConfiguredLanguage(): void
+    {
+        $langRepo = $this->getContainer()->get('language.repository');
+        $languages = $langRepo->search(new Criteria(), Context::createDefaultContext());
+        $languageId = array_values($languages->getElements())[1]->getId();
+
+        $currencyRepo = $this->getContainer()->get('currency.repository');
+        $currencies = $currencyRepo->search(
+            (new Criteria())->addFilter(new EqualsFilter('isoCode', 'EUR')),
+            Context::createDefaultContext()
+        );
+
+        $salesChannelContext = $this->buildNewSalesChannelContext($currencies, $languages);
+        $salesChannelContext->getSalesChannel()->setLanguageId($languageId);
+
+        $this->getContainer()->get('sales_channel.repository')->update([
+            [
+                'id' => $salesChannelContext->getSalesChannel()->getId(),
+                'domains' => [
+                    [
+                        'currencyId' => $currencies->first()->getId(),
+                        'snippetSetId' =>
+                            $salesChannelContext->getSalesChannel()->getDomains()->first()->getSnippetSetId(),
+                        'url' => 'http://cool-url.com/german'
+                    ]
+                ]
+            ],
+        ], Context::createDefaultContext());
+
+        $this->enableFindologicPlugin(
+            $this->getContainer(),
+            self::VALID_SHOPKEY,
```

Comment:
It would be great if there is also a UK channel like described in the comment
> https://some-shop.de/findologic?shopkey=from-united-kingdom => https://shome-shop.co.uk/detail/...

Then I would go with calling http://cool-url.com/german?shopkey=from-united-kingdom and assert english urls are exported, not the /german one.

Reply:
This is just the other way around. The setup for this test is already pretty big. Also since I did test driven development, I know for a fact that this test is exactly testing what it should. The Shopware default Sales Channel is english, so we create a german one and call the english one with the german shopkey => Products are german. I do not see a real benefit of doing this the other way around.

Reply:
Ok understand, when there is an english channel by default, please somehow comment that its using the EN shopkey for the en here.